dali/base/dautils.cpp

```diff
@@ -3625,15 +3627,14 @@ void remapGroupsToDafilesrv(IPropertyTree *file, INamedGroupStore *resolver)
         Owned<IStoragePlane> plane = getDataStoragePlane(planeName, true);
         if ((0 == plane->queryHosts().size()) && isAbsolutePath(plane->queryPrefix())) // if hosts group, or url, don't touch
         {
+            auto updateFunc = [&](const IPropertyTree *oldComponentConfiguration, const IPropertyTree *oldGlobalConfiguration)
             {
                 CriticalBlock b(dafileSrvNodeCS);
-                if (nullptr == dafileSrvNode)
-                {
-                    auto externalService = k8s::getDafileServiceFromConfig("directio");
-                    VStringBuffer dafilesrvEpStr("%s:%u", externalService.first.c_str(), externalService.second);
-                    dafileSrvNode.setown(createINode(dafilesrvEpStr));
-                }
-            }
+                auto externalService = k8s::getDafileServiceFromConfig("directio");
+                VStringBuffer dafilesrvEpStr("%s:%u", externalService.first.c_str(), externalService.second);
+                dafileSrvNode.setown(createINode(dafilesrvEpStr));
```

Comment:
Accessing  dafileSrvNode later on in the pushback needs to (and the createIGroup) needs to be done in a critical section in case it changes inside the call.

Reply:
yes, I'll fix.